### PR TITLE
Handle self links on customresources

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/kubernetes/kubernetes/issues/45776

I started from https://github.com/kubernetes/kubernetes/pull/45777 and wrote the code to make it work properly.

@sdminonne  ptal

### CODE REVIEW

File: staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go [hunk 1]
```diff
@@ -186,3 +186,34 @@ func TestSimpleCRUD(t *testing.T) {
...
+	noxuInstanceToCreate := testserver.NewNoxuInstance(ns, "foo")
+	createdNoxuInstance, err := noxuNamespacedResourceClient.Create(noxuInstanceToCreate)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if e, a := "/apis/mygroup.example.com/v1alpha1/namespaces/not-the-default/noxus/foo", createdNoxuInstance.GetSelfLink(); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+
```
Comment (sttts) [inline staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go]: TODO: add cluster scope test case when clsuter scope custom resources work

**Comment by deads2k** [comment]
@k8s-bot unit test this

File: staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go (hunk 1 above)
Comment (deads2k) [inline staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go]: added

**Review by enisoc — APPROVED** [review]

**Comment by deads2k** [comment]
@k8s-bot unit test this

**Pull request merged**